SolomiiaLomnytska22/solomiia-training
# Added searching by the keywords.

Refs:
#20 (Add filtering to table data)

Proposed changes:
- Added an input field for completing search inside the table
- Added a button for switching filtering options by online/offline values 
- Added error handling by toast notifications

User interface before implementing changes:
<img width="1792" alt="Screenshot 2024-01-17 at 14 32 57" src="https://github.com/SolomiiaLomnytska22/solomiia-training/assets/136481621/38582bb0-0b4e-4c34-9323-430852b1453c">


User interface after implementing changes:
<img width="1576" alt="Screenshot 2024-01-26 at 12 02 16" src="https://github.com/SolomiiaLomnytska22/solomiia-training/assets/136481621/30c59365-2b41-4ef5-8aa9-e7c2b3917e8b">

Verification:
- Open home page with table
- Load data from json server or add new entries using "+ Add" button
- Enter a word of a phrase to find matching entries
- Click button <img width="41" alt="Screenshot 2024-01-17 at 15 55 51" src="https://github.com/SolomiiaLomnytska22/solomiia-training/assets/136481621/b450ad97-865e-4ce6-8d99-a42082ce6f06"> to see your results
- The table will display entries that contain entered word
- To clear your search input press <img width="36" alt="Screenshot 2024-01-17 at 15 57 27" src="https://github.com/SolomiiaLomnytska22/solomiia-training/assets/136481621/ee38b28b-0fa0-469d-b9dd-f7dd4c649224"> button

All changes tested locally. New and already existing functionality is supported.

https://github.com/SolomiiaLomnytska22/solomiia-training/assets/136481621/e64dc50e-85ec-4c53-a63c-cafc2fb60cb9
